src/core/unit.c

```diff
@@ -5373,8 +5373,33 @@ void unit_remove_dependencies(Unit *u, UnitDependencyMask mask) {
         }
 }
 
+static int unit_get_invocation_path(Unit *u, char **path) {
+        char *p;
+
+        assert(u);
+        assert(path);
+
+        if (MANAGER_IS_SYSTEM(u->manager))
+                p = path_join("/run/systemd/units/invocation:", u->id);
+        else {
+                _cleanup_free_ char *user_path = NULL;
+                int r = xdg_user_runtime_dir(&user_path, "/systemd/units/invocation:");
+                if (r < 0) {
+                        log_unit_debug_errno(u, r, "Failed to get user runtime units invocation path: %m");
+                        return r;
```

Comment:
merge these two lines:

```c
if (r < 0)
        return log_unit_debug_errno(u, r, …);
```

i.e. all our log functions with "_errno" in the name return the error you pass as negative value, so that you can merge things like that.